src/driver/amdxdna/amdxdna_mailbox.c

```diff
@@ -378,6 +386,22 @@ static irqreturn_t mailbox_irq_handler(int irq, void *p)
 	return IRQ_HANDLED;
 }
 
+#ifdef AMDXDNA_DEVEL
+static void mbox_timer(struct timer_list *t)
+{
+	struct mailbox_channel *mb_chann = from_timer(mb_chann, t, event_timer);
+
+	if (!mb_chann) {
+		pr_warn("DZ_ mbox is gone, stop timer!\n");
+		return;
+	}
+
```

Comment:
Do we really need this check? If this happen, it looks more like a bug.

For other functions check "!mb_chann" is because they are called from outside of this file. We don't want every caller to check "!mb_chann".

Right before mb_chann is destroied, we must call timer_delete_sync() to wait for this timer function to finish.

Reply:
Thanks, I have fixed this basically removed this checking.